python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -147,10 +151,16 @@ def to_default_asset_events(
         unique_id: str = event_node_info["unique_id"]
         node_resource_type: str = event_node_info["resource_type"]
         node_status: str = event_node_info["node_status"]
+        node_materialization: str = self.raw_event["data"]["node_info"]["materialized"]
 
+        is_node_ephemeral = node_materialization == "ephemeral"
         is_node_successful = node_status == NodeStatus.Success
         is_node_finished = bool(event_node_info.get("node_finished_at"))
-        if node_resource_type in NodeType.refable() and is_node_successful:
+        if (
+            node_resource_type in NodeType.refable()
+            and is_node_successful
+            and not is_node_ephemeral
```

Comment:
confused about this bit -- more along the lines of "how was stuff working before?"

I know we don't create definitions for ephemeral models, so this certainly seems like the correct version of this code, just wondering how we didn't hit errors before this was there.

Reply:
Currently, when we process events from the dbt stream, we only process the `LogXXXResult` events, where `XXX` is one of `Seed`, `Model`, `Test`, etc. These events are always present as they are logged at the info level.

In this PR, we process only the `NodeFinished` events. These `NodeFinished` events are only emitted in debug mode. They essentially have a 1:1 mapping with the `LogXXXResult` events and they have more metadata that is useful to us. 

However, in debug mode, we log `NodeFinished` events for ephemeral models, whereas normally, no events are usually logged at the info level. So now, we need to ignore these ephemeral models.

If you remember https://github.com/dagster-io/dagster/issues/13586, `--debug` used to cause a failure in our integration for this reason -- duplicate events were being produced. Our workaround for this issue was to ignore the debug level events. Now, we can choose to ignore the info level events and process the debug level events!